src/server/game/Handlers/CharacterHandler.cpp

```diff
@@ -987,8 +987,12 @@ void WorldSession::HandlePlayerLoginFromDB(LoginQueryHolder const& holder)
 
     if (pCurrChar->HasAtLoginFlag(AT_LOGIN_CHECK_ACHIEVS))
     {
-        pCurrChar->RemoveAtLoginFlag(AT_LOGIN_CHECK_ACHIEVS, true);
-        pCurrChar->CheckAllAchievementCriteria();
+        // If we process the check while players are loading they won't be notified of the changes.
+        pCurrChar->m_Events.AddEventAtOffset([pCurrChar]
```

Comment:
would we be able to do `if (!pCurrChar->GetSession()->PlayerLoading())` instead of

Reply:
No, because then it would just return false and not run at all 🤷

Reply:
I'm leaving this here because it still fixes achievements not being reported to the client on the first login after flag application